---
title: "Class_19:Investigating_Pertussis_Resurgence"
format: pdf
---

## Investigating Pertussis Resurgence

# Investigating pertussis cases by year

The CDC tracks cases of Pertussis in the US. We can get their data via web-scrapping.

```{r echo=FALSE}
#copy the data and then paste as a data frame by clicking the Addins drop down tab
cdc <- data.frame(
                                 Year = c(1922L,1923L,1924L,1925L,
                                          1926L,1927L,1928L,1929L,1930L,1931L,
                                          1932L,1933L,1934L,1935L,1936L,
                                          1937L,1938L,1939L,1940L,1941L,1942L,
                                          1943L,1944L,1945L,1946L,1947L,
                                          1948L,1949L,1950L,1951L,1952L,
                                          1953L,1954L,1955L,1956L,1957L,1958L,
                                          1959L,1960L,1961L,1962L,1963L,
                                          1964L,1965L,1966L,1967L,1968L,1969L,
                                          1970L,1971L,1972L,1973L,1974L,
                                          1975L,1976L,1977L,1978L,1979L,1980L,
                                          1981L,1982L,1983L,1984L,1985L,
                                          1986L,1987L,1988L,1989L,1990L,
                                          1991L,1992L,1993L,1994L,1995L,1996L,
                                          1997L,1998L,1999L,2000L,2001L,
                                          2002L,2003L,2004L,2005L,2006L,2007L,
                                          2008L,2009L,2010L,2011L,2012L,
                                          2013L,2014L,2015L,2016L,2017L,2018L,
                                          2019L),
         Cases = c(107473,164191,165418,152003,
                                          202210,181411,161799,197371,
                                          166914,172559,215343,179135,265269,
                                          180518,147237,214652,227319,103188,
                                          183866,222202,191383,191890,109873,
                                          133792,109860,156517,74715,69479,
                                          120718,68687,45030,37129,60886,
                                          62786,31732,28295,32148,40005,
                                          14809,11468,17749,17135,13005,6799,
                                          7717,9718,4810,3285,4249,3036,
                                          3287,1759,2402,1738,1010,2177,2063,
                                          1623,1730,1248,1895,2463,2276,
                                          3589,4195,2823,3450,4157,4570,
                                          2719,4083,6586,4617,5137,7796,6564,
                                          7405,7298,7867,7580,9771,11647,
                                          25827,25616,15632,10454,13278,
                                          16858,27550,18719,48277,28639,32971,
                                          20762,17972,18975,15609,18617)
       )
```

```{r}
cdc
```

Make a plot with a trendline

```{r}
library("ggplot2")
library(scales)
```


>Q1. With the help of the R “addin” package datapasta assign the CDC pertussis case number data to a data frame called cdc and use ggplot to make a plot of cases numbers over time.

```{r}
baseplot <- ggplot(cdc, aes(Year, Cases))+
  geom_point()+
  geom_line()+
  labs(title="Cases of Pertussis in US from 1920 to 2019", subtitle="Data from the CDC")+
  scale_y_continuous(labels = label_comma())
```

```{r}
baseplot
```


>Q2. Using the ggplot geom_vline() function add lines to your previous plot for the 1946 introduction of the wP vaccine and the 1996 switch to aP vaccine (see example in the hint below). What do you notice?

```{r}
baseplot+
  geom_vline(xintercept=1946, col="blue")+
  geom_vline(xintercept=1996, col="red")+
  geom_text(aes(x=1944,y=255000,label="wP"))+
  geom_text(aes(x=1994, y=255000, label="aP"))
```

>Q3. Describe what happened after the introduction of the aP vaccine? Do you have a possible explanation for the observed trend?

When the wP vaccine was introduced, case numbers went down since people were getting the vaccine and it was very effective. When the aP vaccine was introduced, the number of cases remained low for a while but began to rise and is on an upward trend. This could be due to a few different reasons such as people being more skeptical to get vaccines due to misinformation or that the aP vaccine isn't as effective as the wP vaccine over a long period of time since cases were low for a while and then began to spike. It could also be that the bacteria evolved and gained some immunity to the vaccine which would explain the later spike in cases. 

# The CMI-PB Project

The CMI=PB project is collecting data on aP and wP individuals and their immune response to infection and/or booster shots. 

CMI-PB returns data from it's API and JSON format (like most APIs). We will use the jsonlite package to get data from this API.

```{r}
library(jsonlite)

subject <- read_json("https://www.cmi-pb.org/api/subject", simplifyVector =T)

head(subject)
```

>Q4. How may aP and wP infancy vaccinated subjects are in the dataset?

```{r}
table(subject$infancy_vac)
```

>Q5. How many Male and Female subjects/patients are in the dataset?

```{r}
table(subject$biological_sex)
```

>Q6. What is the breakdown of race and biological sex (e.g. number of Asian females, White males etc…)?

```{r}
table(subject$race, subject$biological_sex)
```

```{r}
library(lubridate)
```

```{r}
today()
```

>Q7. Using this approach determine (i) the average age of wP individuals, (ii) the average age of aP individuals; and (iii) are they significantly different?

```{r}
age_days <- today() - ymd(subject$year_of_birth)
age_years <- time_length(age_days,"years")
subject$age <-age_years
```

Now find the average age of all individuals:

```{r}
mean(subject$age)
```


Now use splyr to subset to wP or aP subjects

>(i) average age of wP individuals

```{r}
library(dplyr)

ap.age <- filter(subject, infancy_vac =="wP")$age
mean(ap.age)
```


>(ii) average age of aP individuals

```{r}
wp.age <- filter(subject, infancy_vac =="aP")$age
mean(wp.age)
```

> (iii) are they significantly different?

```{r}
t.test(ap.age, wp.age)
```
T-test tells us that the data is significantly different.

>Q8. Determine the age of all individuals at time of boost?

```{r}
int <- ymd(subject$date_of_boost) - ymd(subject$year_of_birth)
age_at_boost <- time_length(int, "year")
head(age_at_boost)
```


>Q9. With the help of a faceted boxplot (see below), do you think these two groups are significantly different?

```{r}
ggplot(subject) +
  aes(time_length(age, "year"),
      fill=as.factor(infancy_vac)) +
  geom_histogram(show.legend=FALSE) +
  facet_wrap(vars(infancy_vac), nrow=2) 
```

Yes they are significantly different, if they were similar their graphs would look much more similar. 

# Joining multiple tables

Read the specimen and ab_titer tables into R and store the data as specimen and titer named data frames.

```{r}
specimen <- read_json("https://www.cmi-pb.org/api/specimen", simplifyVector = T)

titer <- read_json("https://www.cmi-pb.org/api/ab_titer", simplifyVector = T)
```

```{r}
head(specimen)
```

```{r}
head(titer)
```


To know whether a given specimen_id comes from an aP or wP individual we need to link (a.k.a. “join” or merge) our specimen and subject data frames. The excellent dplyr package (that we have used previously) has a family of join() functions that can help us with this common task:


>Q9. Complete the code to join specimen and subject tables to make a new merged data frame containing all specimen records along with their associated subject details:

```{r}
meta <- inner_join(specimen, subject)
dim(meta)
head(meta)
```

>Q10. Now using the same procedure join meta with titer data so we can further analyze this data in terms of time of visit aP/wP, male/female etc.

```{r}
abdata <- inner_join(titer, meta)
dim(abdata)
head(abdata,4)
```

>Q11. How many specimens (i.e. entries in abdata) do we have for each isotype?

```{r}
table(abdata$isotype)
```

>Q12. What do you notice about the number of visit 8 specimens compared to other visits?

```{r}
table(abdata$visit)
```

The number of specimens for 8 visits is significantly lower than the other number of visits. The reason visit 8 is so small is because the project is still ongoing so the data hasn't fully been collected for visit 8 yet. 

# Examine IgG1 Ab titer levels

Exclude visit 8 from the analysis

```{r}
ig1 <- abdata %>% filter(isotype == "IgG1", visit!=8)
head(ig1)
```

>Q13. Complete the following code to make a summary boxplot of Ab titer levels for all antigens:

```{r}
ggplot(ig1) +
  aes(MFI, antigen) +
  geom_boxplot() + 
  facet_wrap(vars(visit), nrow=2)
```

>Q14. What antigens show differences in the level of IgG1 antibody titers recognizing them over time? Why these and not others?

FIM 2/3 (fimbrial protein), FHA (filamentous hemagglutinin), and PRN (pertactin autotransporter). These are rising because they're part of the aP boost vaccine and the immune system is recognizing them. DT also increases as it is one of the antibodies respondign to the bacteria. 
 

We can attempt to examine differences between wP and aP here by setting color and/or facet values of the plot to include infancy_vac status (see below). However these plots tend to be rather busy and thus hard to interpret easily.

```{r}
ggplot(ig1) +
  aes(MFI, antigen, col=infancy_vac ) +
  geom_boxplot(show.legend = FALSE) + 
  facet_wrap(vars(visit), nrow=2) +
  theme_bw()
```

```{r}
ggplot(ig1) +
  aes(MFI, antigen, col=infancy_vac ) +
  geom_boxplot(show.legend = FALSE) + 
  facet_wrap(vars(infancy_vac, visit), nrow=2)
```


>Q15. Filter to pull out only two specific antigens for analysis and create a boxplot for each. You can chose any you like. Below I picked a “control” antigen (“Measles”, that is not in our vaccines) and a clear antigen of interest (“FIM2/3”, extra-cellular fimbriae proteins from B. pertussis that participate in substrate attachment).

```{r}
filter(ig1, antigen=="TT") %>%
  ggplot() +
  aes(MFI, col=infancy_vac) +
  geom_boxplot(show.legend = F) +
  facet_wrap(vars(visit)) +
  theme_bw()
```

```{r}
filter(ig1, antigen=="FIM2/3") %>%
  ggplot() +
  aes(MFI, col=infancy_vac) +
  geom_boxplot(show.legend = T) +
  facet_wrap(vars(visit)) +
  theme_bw()

```

```{r}
filter(ig1, antigen=="PRN") %>%
  ggplot() +
  aes(MFI, col=infancy_vac) +
  geom_boxplot(show.legend = T) +
  facet_wrap(vars(visit)) +
  theme_bw()
```


>Q16. What do you notice about these two antigens time course and the FIM2/3 data in particular?

The TT antigen doesn't increase much over the time course and stays pretty consistent the whole time. For the FIM2/3 data, the boxplot is increasing over the time course and ends at a much higher MFI than the TT. The FIM2/3 peaks around visit 5 and 6 and then starts to decline a bit. 

>Q17. Do you see any clear difference in aP vs. wP responses?

Looking at FIM2/3 and PRN which we saw earlier is involved with the vaccine, the aP response seems to be greater than the wP responses by visit 3 or 4 and then continues to increase at a greater rate than wP and then both the aP and wP vaccine begin to decline around visit 6 or 7. 

# Obtaining CMI-PB RNASeq data

The link is for the key gene involved in expressing any IgG1 antibody, namely the IGHG1 gene. Let’s read available RNA-Seq data for this gene into R and investigate the time course of it’s gene expression values.


```{r}
url <- "https://www.cmi-pb.org/api/v2/rnaseq?versioned_ensembl_gene_id=eq.ENSG00000211896.7"

rna <- read_json(url, simplifyVector = TRUE) 
```

```{r}
head(rna)
```


To facilitate further analysis we need to “join” the rna expression data with our metadata meta, which is itself a join of sample and specimen data. This will allow us to look at this genes TPM expression values over aP/wP status and at different visits (i.e. times):


```{r}
#meta <- inner_join(specimen, subject)
ssrna <- inner_join(rna, meta)
```

>Q18. Make a plot of the time course of gene expression for IGHG1 gene (i.e. a plot of visit vs. tpm).

```{r}
ggplot(ssrna) +
  aes(visit, tpm, group=subject_id) +
  geom_point() +
  geom_line(alpha=0.2)

```

Focus in on visit 4 and facet by aP/wP subjects
```{r}
ssrna %>%  
  filter(visit==4) %>% 
  ggplot() +
    aes(tpm, col=infancy_vac) + geom_density() + 
    geom_rug() 

```


>Q19.: What do you notice about the expression of this gene (i.e. when is it at it’s maximum level)?

The expression of this gene reaches its maximum level at visit 4 and then immediately drops back down to 0 by visit 5.


>Q20. Does this pattern in time match the trend of antibody titer data? If not, why not?

This pattern in time does not match the trend of antibody titer data because this pattern has the maximum level being reached at visit 4 whereas the trend of antibody titer data seems to reach the maximum level around visit 5 and 6. Also, this pattern has the levels going from 0 visit 3 to max at visit 4 and then back to 0 by visit 5. With the antibody titer data, the change among visits was much more gradual. 






